### ISSUE TITLE
Prevent swDest file from being included in precache manifest

### ISSUE THREAD

**Issue by kaycebasques** [GoogleChrome/workbox#647]
I did a lazy glob pattern that matches everything, including my custom SW file and the workbox-sw lib.

Any problem with that? It seems to work and update fine, but something seems dicey about this...

**Comment by prateekbh** [comment]
>It seems to work and update fine

I doubt this, if service worker caches itself, it might always be returned from cacheFirst. Which would prevent your web app from updating ever.

**Comment by jeffposnick** [comment]
The service worker specification states that when a browser checks the registered `sw.js` file for an update, it shouldn't trigger the `fetch` handler of the controlling service worker. So it doesn't matter from an updates perspective if your service worker precaches itself.

(HTTP `Cache-Control` headers can screw things up, but that's independent from the service worker's `fetch` handler.)

That being said, having a service worker cache its own JS does waste bandwidth and storage space, and I thought we had some protections in `workbox-build` to prevent the `swDest` file from being added to the precache manifest. Could you pass along your `workbox-build` configuration and we can see what's going wrong?

**Comment by kaycebasques** [comment]
I'm just doing a lazy glob that matches everything in my output directory

```
function buildSw() {
  return wb.injectManifest({
    globDirectory: './_site/',
    globPatterns: ['**\/*'],
    swSrc: './src/sw.js',
    swDest: './_site/sw.js',
  }).then(() => {
    gutil.log('Build Manifest generated.');
  }).catch((error) => {
    gutil.log('Error: ' + error);
  });
}
```

**Comment by gauntface** [comment]
@kaycebasques could you create a zip of your project and email it to me? There are some checks in place for this kind of stuff, but wouldn't surprise me if I have still messed something up.

**Comment by gauntface** [comment]
Pinging @kaycebasques Could you give this another try with 1.1.0 now that it's released and if it's still happening could you send me your example project.

**Comment by gauntface** [comment]
I'm going to close this as I believe it's fixed in 1.1.0.

If anyone hits this bug PLEASE PLEASE PLEASE send me an example project <3
